app/javascript/mastodon/features/audio/index.js

```diff
@@ -336,11 +336,16 @@ class Audio extends React.PureComponent {
     const AudioContext = window.AudioContext || window.webkitAudioContext;
     const context      = new AudioContext();
     const source       = context.createMediaElementSource(this.audio);
+    const gainNode     = context.createGain();
+
+    gainNode.gain.value = this.audio.volume;
```

Comment:
This doesn't sound right, if `gainNode.gain.value` is supposed to replace `this.audio.volume` and `this.audio.volume` can be applied on top of `gainNode.gain.value`.

There are also still references to `this.audio.muted` on lines 84, 118 and 327, as well as references to `this.audio.volume` on lines 83, 118, and 323.

Reply:
Thank you so much, @ClearlyClaire. You perfectly pointed out what I was missing.